code-challenges.test.js

```diff
@@ -20,30 +20,160 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
 
 // a) Create a test with expect statements for each of the variables provided.
 
+// const nums1 = [3, 56, 90, -8, 0, 23, 6]
+// // Expected output: [-8, 90]
+// const nums2 = [109, 5, 9, 67, 8, 24]
+// // Expected output: [5, 109]
+
+describe("minMax", () => {
+
+  it("takes an array of numbers and returns an array of the minimum and maximum numbers in that order", () => {
+    const nums1 = [3, 56, 90, -8, 0, 23, 6]
+    // Expected output: [-8, 90]
+    const nums2 = [109, 5, 9, 67, 8, 24]
+    // Expected output: [5, 109]
+    expect(minMax(nums1)).toEqual([-8, 90])
+    expect(minMax(nums2)).toEqual([5, 109])
+  })
+})
+
+// ReferenceError: minMax is not defined :)
+
+// b) Create the function that makes the test pass.
+
+// PC:
+// create function named minMax
+  // takes an array
+// create empty placeholder array
+// use Math.min to find lowest value
+  // .push() to empty array
+// use Math.max to find highest value
+  // .push() to empty array 
+// for loop to iterate through array
+// .filter out bad values
+// return placeholder array
+
 const nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 const nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
 
+function minMax(array){
+  var a = []
+  for(i = 0; i < array.length; i++) {
+    a.push(Math.min(array));
+    a.push(Math.max(array));
+  }
+  let b = a.filter(number => number)
+  return b
+}
 
-// b) Create the function that makes the test pass.
 
+// Results so far:
+
+// expect(received).toEqual(expected) // deep equality
+
+// - Expected  - 4
+// + Received  + 1
+
+// - Array [
+// -   -8,
+// -   90,
+// - ]
+// + Array []
+
+// // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+
+// // a) Create a test with an expect statement using the variables provided.
 
-// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+describe("dblArrayFn", () => {
 
-// a) Create a test with an expect statement using the variables provided.
+  it("that takes in two arrays as arguments and returns one array with no duplicate values", () => {
+    const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+    const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+    expect(dblArrayFn(testArray1, testArray2)).toEqual(expect.arrayContaining([3, 7, 10, 5, 4, 8, 2, 1]))
+  })
+})
+
+// GOOD FAIL:  ReferenceError: dblArrayFn is not defined
+
+// PC:
+// create fn names dblArrayFn
+  // takes in 2 array
+// concat testArray1 to testArray2
+// sort the array
+// for loop over new array 
+  // check for duplicate values with iteration
+    // if value === value at [i], filter it out
+// return array
+
+// // b) Create the function that makes the test pass
```

Comment:
I am definitely seeing an improvement. Glad that you attempted this question. Please review this link to help you find a solution you understand. https://levelup.gitconnected.com/7-ways-to-remove-duplicates-from-array-in-javascript-cea4144caf31